### DISCUSSION TITLE
redeployment bug

### DISCUSSION THREAD

**Discussion by hajpepe** [vercel/next.js#49458] [Help] [answered]
### Summary

Dear Vercel team,

I'm experiencing an error when trying to redeploy my project. The error appears in the new commit, while the first deployment did not show this error. I have successfully built the project locally without any issues.

The error message includes the following warnings:

> Warning: React Hook React.useEffect has a missing dependency: 'currentVariant'. Either include it or remove the dependency array. react-hooks/exhaustive-deps
> Warning: React Hook useEffect has a missing dependency: 'currentVariant'. Either include it or remove the dependency array. react-hooks/exhaustive-deps
> In addition to the warnings, the deployment failed with the following message:
> 
> Module not found: Can't resolve 'components/loading/loading.jsx' in '/vercel/path0/components/Index'
> Error: Command "npm run build" exited with 1

### I would appreciate your help in resolving this issue. The GitHub repository for this project can be found here: https://github.com/hajpepe/AfshinFX

Thank you for your assistance.

Best regards

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I think the issue is case-sensitivity in the path and file names, for example here: https://github.com/hajpepe/AfshinFX/blob/5c94197aca89587403c46c5781bf1aac3ec4e28f/components/Index/Index.jsx#L11

You import:

```js
import Loading from 'components/loading/loading.jsx';
```

However, there's no such path. Instead, what you have is: `components/Loading/Loading.jsx`

> You don't really need the file extension either

What happens is that locally, you are probably using Windows, which IIRC is insensitive to casing, so it sees `components/Loading/Loading.jsx` and `components/loading/loading.jsx` as the same, whereas, Linux (where Vercel hosts your project), sees them as different paths. 

https://learn.microsoft.com/en-us/windows/wsl/case-sensitivity


> - Windows file system treats file and directory names as case-insensitive. FOO.txt and foo.txt will be treated as equivalent files.
> - Linux file system treats file and directory names as case-sensitive. FOO.txt and foo.txt will be treated as distinct files.

So you should have:

```js
import Loading from 'components/Loading/Loading';
```


In your case, you'll also need to tell Git that you are changing the casing, https://stackoverflow.com/a/20907647 -

**Reply by hajpepe** [reply]
hello Joseph , 
before i change that to 
loading.jsx 
in this commit you can see components/loading/loading without jsx :
https://github.com/hajpepe/AfshinFX/commit/4a2635130e54684adbf3324f74a04986f12d2c39

**Reply by icyJoseph** [reply]
Yeah but the `.jsx`is not the problem, the core problem is the casing. 

`Loading` and `loading` is not the same for a Linux file system, such as the one that hosts your application in the Vercel platform. 

Locally your OS (most likely windows) sees them as the same. Change your paths to match what you have.

**Reply by icyJoseph** [reply]
I have edited my answer to make it more clear.

**Reply by hajpepe** [reply]
thank youuu :)
